.github/workflows/kubectl-az.yml

```diff
@@ -0,0 +1,142 @@
+          ${{ runner.os }}-go-
+    - name: Check out code
+      uses: actions/checkout@v2
+    - name: Build and generate tarball
+      run: |
+        target=kubectl-az-${{ matrix.os }}-${{ matrix.arch }}
+
+        make $target
+
+        # Prepare binary as artifact, it will be used by other jobs
+        mv $target kubectl-az
+        tar --sort=name --owner=root:0 --group=root:0 \
+          -czf ${target}.tar.gz \
+          kubectl-az LICENSE
+    - name: Add kubectl-az-${{ matrix.os }}-${{ matrix.arch }}.tar.gz as artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: kubectl-az-${{ matrix.os }}-${{ matrix.arch }}
+        path: kubectl-az-${{ matrix.os }}-${{ matrix.arch }}.tar.gz
+
+  unit-tests:
+    name: Run unit tests
+    runs-on: ubuntu-latest
+    steps:
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ env.GO_VERSION }}
+        check-latest: true
+    - name: Cache Go
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.cache/go-build
+          ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+    - name: Check out code
+      uses: actions/checkout@v2
+    - name: Run tests
+      run: make unit-test
+
+  integration-tests:
+    name: Run integration tests
+    needs: [build, unit-tests]
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macOS-latest, windows-latest]
+        arch: [amd64] # TODO: Support ARM
+    steps:
+    - name: Set environment variables
+      shell: bash
+      run: |
+        case ${{ matrix.os }} in
+          ubuntu-latest)
+            echo "os=linux" >> $GITHUB_ENV
+            ;;
+          macOS-latest)
+            echo "os=darwin" >> $GITHUB_ENV
+            ;;
+          windows-latest)
+            echo "os=windows" >> $GITHUB_ENV
+            ;;
+          *)
+            echo "Not supported OS: ${{ matrix.os }}"
+            exit 1
+            ;;
+        esac
+    - name: Get kubectl-az from artifact
+      uses: actions/download-artifact@v2
+      with:
+        name: kubectl-az-${{ env.os }}-${{ matrix.arch }}
+    - name: Prepare kubectl-az binary
+      shell: bash
+      run: |
+        tar zxvf kubectl-az-${{ env.os }}-${{ matrix.arch }}.tar.gz
+        chmod +x kubectl-az
+    - name: Authenticate and set AKS cluster context
+      # TODO: Create a cluster instead of using an existing one
+      # Required to allow kubectl-az to communicate with Kubernetes API server
+      # It is equivalent to running 'az aks get credentials'
+      uses: azure/aks-set-context@v1
+      env:
+        CLUSTER_NAME: "kubectl_az_ci_cluster"
+        CLUSTER_RESOURCE_GROUP: "kubectl-az-ci-rg"
+      with:
+        creds: '${{ secrets.AZURE_CREDENTIALS }}'
+        cluster-name: ${{ env.CLUSTER_NAME }}
+        resource-group: ${{ env.CLUSTER_RESOURCE_GROUP }}
```

Comment:
Probably, we could create just one cluster and test all the binaries against it.

Reply:
Done